### ISSUE TITLE
Allow middlewares to be registered on a single route

### ISSUE THREAD

**Issue by opensaucerer** [opensaucerer/barf#9]
Currently, BARF supports hijacking all routes and single subrouters via middleware injection as give below
```go
// apply to all routes
barf.Hippocampus().Hijack()

r := barf.RetroFrame("/path")
// apply to the subrouter r
barf.Hippocampus(r).Hijack()
```

We need to make it possible to also add support for adding middleware to just one route such that the middleware is only called when that route is matched. The pattern would be of the form
```go
func Get(path string, handler func(http.ResponseWriter, *http.Request), m ...typing.Middleware)
```

**Comment by oxiginedev** [comment]
I have an idea of how this might be possible, could I make a draft for you to go through?

**Comment by opensaucerer** [comment]
> I have an idea of how this might be possible, could I make a draft for you to go through?

For sure...I just assigned this to you.

**Comment by opensaucerer** [comment]
Resolved by #12
